src/Repository/ElectedRepresentative/MandateRepository.php

```diff
@@ -74,4 +75,24 @@ public function findByFunctionAndUserListDefinitionForAdherent(
             ->getResult()
         ;
     }
+
+    public function hasMayorMandate(Adherent $adherent): bool
+    {
+        $mandates = $this
+            ->createQueryBuilder('mandate')
+            ->leftJoin('mandate.electedRepresentative', 'electedRepresentative')
+            ->leftJoin('mandate.politicalFunctions', 'politicalFunction')
+            ->andWhere('electedRepresentative.adherent = :adherent')
+            ->andWhere('politicalFunction.name = :mayor')
+            ->andWhere('mandate.isElected = 1')
+            ->andWhere('mandate.onGoing = 1')
+            ->andWhere('politicalFunction.onGoing = 1')
+            ->setParameter('adherent', $adherent)
+            ->setParameter('mayor', TerritorialCouncilQualityEnum::MAYOR)
+            ->getQuery()
+            ->getResult()
+        ;
+
+        return \count($mandates) > 0;
```

Comment:
it's better to do a `SELECT COUNT(1)` in this case and avoid the doctrine hydration process:

```php
return 0 < (int) $this->createQueryBuilder()
  ->select(`COUNT(1)`)
  ...
  ->getSingleScalarResult()
```